src/celengine/astroobj.h

```diff
@@ -21,7 +21,22 @@ class AstroObject
     AstroCatalog::IndexNumber getIndex() const { return m_mainIndexNumber; }
     void setIndex(AstroCatalog::IndexNumber);
 
+//  Auto Indexing stuff
+    static constexpr AstroCatalog::IndexNumber MaxAutoIndex = UINT32_MAX - 1;
+    static AstroCatalog::IndexNumber getNewAutoIndex() { return m_autoIndex--; }
+    static AstroCatalog::IndexNumber getCurrentAutoIndex() { return m_autoIndex; }
+    static void recoverAutoIndex()
+    {
+        if (m_autoIndex < MaxAutoIndex)
+            ++m_autoIndex;
+    }
+    AstroCatalog::IndexNumber setNewAutoIndex();
+    static void resetAutoIndex(AstroCatalog::IndexNumber i = MaxAutoIndex) { m_autoIndex = i; }
```

Comment:
`reset` means `return to initial state`

Reply:
I though it's good enaugh and dont interferes with `setAutoIndex`. Again, I've trouble with finding very accurate and concrete names.